javatests/build/bazel/tests/integration/BazelBaseTestCaseTest.java

```diff
@@ -14,19 +14,201 @@
+                            .appendValue(exitCode)
+                            .appendText("\n")
+                            .appendText("Workspace contents: \n")
+                            .appendValueList("","\n","\n", contents(workspace.toPath()))
+                            .appendDescriptionOf(commandDescription(cmd));
+                    return false;
+                }
+                return true;
+            }
+
+            @Override
+            public void describeTo(final Description description) {
+               description.appendText("successful exit code (0)");
+            }
+        };
+    }
+
+    private SelfDescribing commandDescription(final Command cmd) {
+        return description -> {
+            final String newLine = System.getProperty("line.separator");
+            final List<String> logContents = logsOfInternalTests(cmd.getErrorLines()).collect(Collectors.toList());
+            description
+                    .appendText("std-error:\n")
+                    .appendValueList("", newLine, newLine,cmd.getErrorLines());
+            if (!logContents.isEmpty()) {
+            description
+                    .appendText("Contents of internal test logs:\n")
+                    .appendText("*******************************\n")
+                    .appendValueList(newLine, newLine, newLine,
+                            logContents);
+            }
+        };
+    }
+
+    private List<String> contents(final Path workspacePath) {
+        try {
+            return Files.walk(workspacePath).map(Path::toString).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Stream<String> logsOfInternalTests(final List<String> errorLines) {
+        return errorLines.stream()
+                .filter(line -> line.contains("(see "))
+                .map(line -> line.split("see ")[1].replace(")", ""))
+                .map(Paths::get)
+                .map(logPath -> {
+                    try {
+                        LinkedList<String> logContents = new LinkedList<>(Files.readAllLines(logPath));
+                        logContents.addFirst("Log contents:");
+                        logContents.addFirst(logPath.toString());
+                        logContents.addFirst("Log path:");
+                        return logContents;
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .flatMap(Collection::stream);
+    }
+
+    private void setupPassingTest(final String testName) throws IOException {
+        writePassingTestJavaSource(testName);
+        writeTestBuildFile(testName);
+    }
+
+    private void writeTestBuildFile(final String testName) throws IOException {
+        scratchFile("BUILD", "load(\"//:bazel_integration_test.bzl\", \"bazel_java_integration_test\")\n" +
```

Comment:
You can use single quote to avoid backslash and `", "` instead of `\n" + "`, the scratchFile joins the lines

Reply:
done. Needed some tricks to concat lists and individual strings so take a peak to see if you like